File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    days = []
    
    for p,s in zip(progresses, speeds):
        days.append([p,math.ceil((100-p)/s)])
    
    days = days[::-1]
    print(days)
    now = days.pop()[1]
    cnt = 1
    while days:
        tmp = days.pop()[1]
        if now >= tmp:
            cnt += 1
        else:
            answer.append(cnt)
            cnt = 1
            now = tmp
    answer.append(cnt)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> result = new ArrayList<>();
        
        int n = progresses.length;
        int curDay = 0; int count = 0;
        
        for (int i=0;i<n;i++) {
            int remain = 100 - progresses[i];
            int day = (remain + speeds[i] - 1) / speeds[i];
            
            if (count == 0) {
                curDay = day;
                count = 1;
            } else if (day <= curDay) {
                count ++;
            } else {
                result.add(count);
                curDay = day;
                count = 1;
            }
        }
        
        if (count > 0) result.add(count);
        
        int[] answer = new int[result.size()];
        for (int i=0;i<result.size();i++) answer[i] = result.get(i);
        
        return answer;
    }
}